rpc_util.go

```diff
@@ -874,14 +874,35 @@ func decompress(compressor encoding.Compressor, d mem.BufferSlice, maxReceiveMes
 		return nil, 0, err
 	}
 
-	out, err := mem.ReadAll(io.LimitReader(dcReader, int64(maxReceiveMessageSize)+1), pool)
+	out, err := mem.ReadAll(io.LimitReader(dcReader, int64(maxReceiveMessageSize)), pool)
 	if err != nil {
 		out.Free()
 		return nil, 0, err
 	}
+	if err = checkReceiveMessageOverflow(int64(out.Len()), int64(maxReceiveMessageSize), dcReader); err != nil {
```

Comment:
With the current handling in `recvAndDecompress` we can't return a non-nil error here. If we do, the returned status code will become `Internal` instead of `ResourceExhausted` resulting in a behaviuor change. I've suggested an alternate below.

Reply:
Done